cmd/lotus-miner/proving.go

```diff
@@ -640,7 +641,42 @@ It will not send any messages to the chain.`,
 		if err != nil {
 			return err
 		}
-		jr, err := json.Marshal(res)
+
+		//convert sector information into easily readable information
+		type PoStPartition struct {
+			Index   uint64
+			Skipped []uint64
+		}
+		type SubmitWindowedPoStParams struct {
+			Deadline         uint64
+			Partitions       []PoStPartition
+			Proofs           []proof.PoStProof
+			ChainCommitEpoch abi.ChainEpoch
+			ChainCommitRand  abi.Randomness
+		}
+		var postParams []SubmitWindowedPoStParams
+		for _, i := range res {
+			var postParam SubmitWindowedPoStParams
+			postParam.Deadline = i.Deadline
+			for id, part := range i.Partitions {
+				postParam.Partitions[id].Index = part.Index
+				count, err := part.Skipped.Count()
+				if err != nil {
+					return err
+				}
+				sectors, err := part.Skipped.All(count)
+				if err != nil {
+					return err
+				}
+				postParam.Partitions[id].Skipped = sectors
+			}
+			postParam.Proofs = i.Proofs
+			postParam.ChainCommitEpoch = i.ChainCommitEpoch
+			postParam.ChainCommitRand = i.ChainCommitRand
+			postParams = append(postParams, postParam)
+		}
+
+		jr, err := json.Marshal(postParams)
```

Comment:
I would just output in a human readable format. This was outputting json because that was the simplest way to have some useful debug output.

Reply:
Sorry, I didn't understand what you meant. You mean it's not easy enough to use is it now?

Reply:
I think this is enough, if there is a more elegant way, please suggest it and I will learn.
```
lotus-miner proving compute window-post 0|grep -v Took|jq .
or
lotus-miner proving compute window-post 0|grep -v Took|jq .[].Partitions[].Skipped
```

Reply:
It is easy now, but I'm not sure if the output should be json as most commands output data in more human-readable form (and some commands have --json for json output)

e.g.
```
$ lotus-miner sectors status 1715                                                                                                                                                                                                                                          SectorID:	1715
Status:		Proving
CIDcommD:	baga6ea4seaqao7s73y24kcutaosvacpdjgfe5pw76ooefnyqw4ynr3d2y6x2mpq
[...]
```

I guess at least this json could be output with json.MarshalIndent.

Note that this is purely for consistency with other commands, there is nothing technically wrong with this approach.

Reply:
ok,I agree with standardization the output.